### PR TITLE
Default Country Bug

### PR DESCRIPTION
**Pull request by harimohanraj89**
Story: https://www.pivotaltracker.com/story/show/170972119

This PR addresses a bug where the edit facility screen would always pre-populate the "Country" field to be "India". The following changes are included:

* Only pre-populate a country value if none exists
* Pre-populate a country based on configuration, instead of always populating "India"
* Create a `Rails.application.config.country` hash that stores country-specific configuration in memory
* Consolidate all time-zone configurations into a single env var
* Add support for using a gitignored `.env.development.local` to override `.env.development` configurations locally.

### CODE REVIEW

File: .env.test [hunk 1]
```diff
@@ -93,11 +93,7 @@ SIMPLE_SERVER_HOST_PROTOCOL='https'
 #
 # Country specific configuration
 #
-ANALYTICS_TIME_ZONE='Asia/Kolkata'
-CRON_TIME_ZONE='Asia/Kolkata'
 DEFAULT_COUNTRY=IN
```
Comment (govindkrjoshi) [inline .env.test]: Shall we make this full country name to avoid ambiguity?

File: config/initializers/countries.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  },
+  'BD': {
+    abbreviation: 'BD',
+    name: 'Bangladesh',
+    dashboard_locale: ENV['DEFAULT_PREFERRED_DASHBOARD_LOCALE'] || 'en_BD',
+    time_zone: ENV['DEFAULT_TIME_ZONE'] || 'Asia/Dhaka'
+  }
+}.with_indifferent_access
+
+Rails.application.config.country = country_config[ENV.fetch('DEFAULT_COUNTRY')]
```
Comment (govindkrjoshi) [inline config/initializers/countries.rb]:
```suggestion
Rails.application.config.default_country = country_config[ENV.fetch('DEFAULT_COUNTRY')]
```

File: config/initializers/countries.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+country_config = {
+  'IN': {
+    abbreviation: 'IN',
+    name: 'India',
+    dashboard_locale: ENV['DEFAULT_PREFERRED_DASHBOARD_LOCALE'] || 'en_IN',
```
Comment (govindkrjoshi) [inline config/initializers/countries.rb]: These values should be country specific constants, right?

File: .env.test (hunk 1 above)
Comment (harimohanraj89) [reply .env.test]: I'm ok leaving this as a country code. The code is still being used in some places too (specifically phone number parsing, I think).

File: config/initializers/countries.rb (hunk 2 above)
Comment (harimohanraj89) [reply config/initializers/countries.rb]: Why do we use the `default` terminology? Can an instance of Simple switch between countries effectively? In other words, can a single instance of Simple show reports in IST _and_ also in Bangla time?

File: config/initializers/countries.rb (hunk 3 above)
Comment (harimohanraj89) [reply config/initializers/countries.rb]: Yes. The second values in these statements are the country-specific ones. The env var just allows us to override the country-specific values _if_ we need to. I suspect that we won't need to do this any time soon.

**Pull request merged**